main/src/library/BagMultiMap.flix

```diff
@@ -0,0 +1,595 @@
+        sumWith((k, _) -> k, m)
+
+    ///
+    /// Returns the sum of all values in the BagMultiMap `m`.
+    ///
+    pub def sumValues(m: BagMultiMap[k, Int32]): Int32 =
+        sumWith((_, v) -> v, m)
+
+    ///
+    /// Returns the sum of all key-value pairs `k => v` in the BagMultiMap `m` according to the function `f`.
+    ///
+    /// Purity reflective: Runs in parallel when given a pure function `f`.
+    ///
+    @ParallelWhenPure
+    pub def sumWith(f: (k, v) -> Int32 & ef, m: BagMultiMap[k, v]): Int32 & ef =
+        let BagMultiMap(m1) = m;
+        let outer = (k, vs) -> Nec.sumWith(v -> f(k, v), vs);
+        Map.sumWith(outer, m1)
+
+    ///
+    /// Returns the product of all keys in the BagMultiMap `m`.
+    ///
+    pub def productKeys(m: BagMultiMap[Int32, v]): Int32 =
+        productWith((k, _) -> k, m)
+
+    ///
+    /// Returns the product of all values in the BagMultiMap `m`.
+    ///
+    pub def productValues(m: BagMultiMap[k, Int32]): Int32 =
+        productWith((_, v) -> v, m)
+
+    ///
+    /// Returns the product of all key-value pairs `k => v` in the BagMultiMap `m` according to the function `f`.
+    ///
+    /// Purity reflective: Runs in parallel when given a pure function `f`.
+    ///
+    @ParallelWhenPure
+    pub def productWith(f: (k, v) -> Int32 & ef, m: BagMultiMap[k, v]): Int32 & ef =
+        let BagMultiMap(m1) = m;
+        let outer = (k, vs) -> Nec.productWith(v -> f(k, v), vs);
+        Map.productWith(outer, m1)
+
+    ///
+    /// Returns `true` if and only if at least one mapping in `m` satisfies the predicate `f`.
+    ///
+    /// Returns `false` if `m` is the empty BagMultiMap.
+    ///
+    /// Purity reflective: Runs in parallel when given a pure function `f`.
+    ///
+    @ParallelWhenPure
+    pub def exists(f: (k, v) -> Bool & ef, m: BagMultiMap[k, v]): Bool & ef =
+        let BagMultiMap(m1) = m;
+        let outer = (k, vs) -> Nec.exists(v -> f(k, v), vs);
+        Map.exists(outer, m1)
+
+    ///
+    /// Returns `true` if and only if all mappings in `m` satisfy the predicate `f`.
+    ///
+    /// Returns `true` if `m` is the empty BagMultiMap.
+    ///
+    /// Purity reflective: Runs in parallel when given a pure function `f`.
+    ///
+    @ParallelWhenPure
+    pub def forall(f: (k, v) -> Bool & ef, m: BagMultiMap[k, v]): Bool & ef =
+        let BagMultiMap(m1) = m;
+        let outer = (k, vs) -> Nec.forall(v -> f(k, v), vs);
+        Map.forall(outer, m1)
+
+    ///
+    /// Returns the union of `m1` and `m2`.
+    ///
+    /// If a key is common to both BagMultiMaps then the left and right bags are concatenated.
+    ///
+    pub def union(m1: BagMultiMap[k, v], m2: BagMultiMap[k, v]): BagMultiMap[k, v] with Order[k] =
+        let BagMultiMap(mm1) = m1;
+        let BagMultiMap(mm2) = m2;
+        BagMultiMap(Map.unionWith((n1, n2) -> Nec.append(n1, n2), mm1, mm2))
+
+    ///
+    /// Returns the intersection of `m1` and `m2`.
```

Comment:
A quick explanation of what `intersection` means for `BagMultiMap`s would be helpful

Reply:
Fixed